src/main/kotlin/no/nav/familie/ks/sak/kjerne/adopsjon/AdopsjonService.kt

```diff
@@ -0,0 +1,50 @@
+package no.nav.familie.ks.sak.kjerne.adopsjon
+
+import jakarta.transaction.Transactional
+import no.nav.familie.ks.sak.common.BehandlingId
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Vilkår
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class AdopsjonService(
+    private val adopsjonRepository: AdopsjonRepository,
+) {
+    fun hentAlleAdopsjonerForBehandling(behandlingId: BehandlingId): List<Adopsjon> = adopsjonRepository.hentAlleAdopsjonerForBehandling(behandlingId.id)
+
+    @Transactional
+    fun oppdaterAdopsjonsdato(
+        behandlingId: BehandlingId,
+        aktør: Aktør,
+        nyAdopsjonsdato: LocalDate?,
+        vilkår: Vilkår,
+    ) {
+        if (vilkår == Vilkår.BARNETS_ALDER) {
+            val nåværendeAdopsjon = adopsjonRepository.finnAdopsjonForAktørIBehandling(behandlingId = behandlingId.id, aktør = aktør)
+
+            if (nåværendeAdopsjon?.adopsjonsdato != nyAdopsjonsdato) {
+                if (nåværendeAdopsjon != null) {
+                    adopsjonRepository.delete(nåværendeAdopsjon)
+                    adopsjonRepository.flush()
+                }
+                if (nyAdopsjonsdato != null) {
+                    adopsjonRepository.saveAndFlush(Adopsjon(behandlingId = behandlingId.id, aktør = aktør, adopsjonsdato = nyAdopsjonsdato))
+                }
+            }
+        }
+    }
```

Comment:
Hvis nyAdopsjonsdato som er sendt inn her er null, så vil man slette nåværende adopsjonsobjekt og ikke lage ny.
Er det meningen? 

Bare lurer på om man skal ha en spesifikk slett adopsjon metode isåfall 🤔